compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1873,6 +1873,66 @@ static FailureOr<Operation *> getRootOperation(
   return nullptr;
 }
 
+static LogicalResult adjustTileSizesForPackOp(func::FuncOp entryPointFn,
+                                              Operation *rootOp) {
+  // The check is not needed if we land
+  // https://github.com/openxla/iree/pull/12594
+  if (!rootOp) return success();
+
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(rootOp);
```

Comment:
Why linalg op? Any root op is fine right?

Reply:
The dimensions could be interchanged or broadcasted. This make sure that we'll track on right dimensions.

Reply:
I think we should just take the first level tile size and use is as is. At worst we switch the tile sizes across dimensions. That shouldnt have too much effect for now. This kind of tracking gets complicated very quickly. We can revisit this if it becomes an issue.

Reply:
I agree that this does not have effort for now. I addressed the comment, but I'd like to point out that it only save us 4-5 lines and logic is not that complicated, see https://github.com/openxla/iree/pull/12590/commits/95297fd8a07c36bb27878945312b122882e548ad